db/todo.go

```diff
@@ -0,0 +1,41 @@
+package db
+
+import (
+	"github.com/google/uuid"
+	"github.com/transactions/entity"
+	"gorm.io/gorm"
+)
+
+type todoDB struct {
+	db *gorm.DB
+}
+
+type TodoDB interface {
+	GetByID(id uuid.UUID) (entity.Todo, error)
+}
+type TrxTodoDB interface {
+	TrxUpdate(tx *gorm.DB, todo entity.Todo) error
+	TrxCreate(tx *gorm.DB, todo entity.Todo) error
+}
+
+func NewTodoDB(db *gorm.DB) *todoDB {
+	return &todoDB{db: db}
+}
+
+func (t *todoDB) GetByID(id uuid.UUID) (entity.Todo, error) {
+	var todo entity.Todo
+	err := t.db.Table("todos").Preload("Subs").First(&todo, "id = ?", id.String()).Error
+	return todo, err
+}
+
+func (t *todoDB) TrxUpdate(tx *gorm.DB, todo entity.Todo) error {
```

Comment:
The type of a transaction is "gorm.db". You can pass this transaction obj as an argument in the constructor of todoDB. 
You can call this transaction like `t.db.Create`.
I have done it by creating new repo objects in every unit of work call, but you can just create one repo method WithTrx(tx) that returns repo obj with transaction in field
Pro:
 - You hide implementation details in repository level removing Trx from name and argument list

Reply:
of course i can use WithTrx method but i tried to avoid that, to be honest i don't know best way to do that, i'm waiting for other friend to review and i hope we will resolve the problem )

Reply:
Maybe we can add methods for starting transaction to a db repo as an composite interface like
Begin() DB
Commit DB
Rollback() DB